packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -312,6 +322,42 @@ describe('<DataGrid /> - Filter', () => {
       });
     });
 
+    ['contains', 'startsWith', 'equals', 'endsWith'].forEach((operatorValue) => {
+      it(`should show all rows when the value is '' and operator='${operatorValue}'`, () => {
+        render(
+          <TestCase
+            filterModel={undefined}
+            columns={[{ field: 'brand' }]}
+            rows={[
+              {
+                id: 3,
+                brand: 'Asics',
+              },
+              {
+                id: 4,
+                brand: 'RedBull',
+              },
+              {
+                id: 5,
+                brand: 'Hugo',
+              },
+            ]}
+          />,
+        );
+        expect(getColumnValues()).to.deep.equal(['Asics', 'RedBull', 'Hugo']);
+        fireEvent.change(screen.getByRole('combobox', { name: 'Operators' }), {
+          target: { value: operatorValue },
+        });
+        const input = screen.getByRole('textbox', { name: 'Value' });
+        fireEvent.change(input, { target: { value: 'abc' } });
```

Comment:
Instead of using the UI controls, did you consider using the control prop for the filter? Maybe this will be more resilient to future UI changes? For example as in https://codesandbox.io/s/basiceditinggrid-material-demo-forked-ttl15?file=/demo.tsx

Reply:
We need a balance between testing the whole user interaction and using the control prop. For instance, #2940 was caused because we lacked testing the first. In theory we should be testing both. I don't know if we should have an exact % of each category of test.